src/SBInclinedSersic.cpp

```diff
@@ -0,0 +1,469 @@
+                           _r0 = _info->calculateScaleForTruncatedHLR(_re, _trunc);
+                       }
+
+                       // Update _info with the correct truncated version.
+                       _info = cache.get(boost::make_tuple(_n,_trunc/_r0,
+                                                           this->gsparams.duplicate()));
+
+                       if (flux_untruncated) {
+                           // Update the stored _flux and _re with the correct values
+                           _flux *= _info->getFluxFraction();
+                           _re = _r0 * _info->getHLR();
+                       }
+                   } else {
+                       // Then given HLR and flux are the values for the untruncated profile.
+                       _r0 = _re / _info->getHLR();
+                   }
+               }
+               break;
+          case SCALE_RADIUS:
+               {
+                   _r0 = size;
+                   if (_truncated) {
+                       // Update _info with the correct truncated version.
+                       _info = cache.get(boost::make_tuple(_n,_trunc/_r0,
+                                                           this->gsparams.duplicate()));
+
+                       if (flux_untruncated) {
+                           // Update the stored _flux with the correct value
+                           _flux *= _info->getFluxFraction();
+                       }
+                   }
+                   // In all cases, _re is the real HLR
+                   _re = _r0 * _info->getHLR();
+               }
+               break;
+          default:
+               throw SBError("Unknown SBInclinedSersic::RadiusType");
+        }
+        dbg<<"hlr = "<<_re<<std::endl;
+        dbg<<"r0 = "<<_r0<<std::endl;
+
+        _inv_r0 = 1./_r0;
+        _half_pi_h_sini_over_r = 0.5*M_PI*_h0*std::abs(_inclination.sin())/_r0;
+
+        dbg << "inv_r0 = " << _inv_r0 << std::endl;
+        dbg << "half_pi_h_sini_over_r = " << _half_pi_h_sini_over_r << std::endl;
+
+        _r0_sq = _r0*_r0;
+        _inv_r0 = 1./_r0;
+        _inv_r0_sq = _inv_r0*_inv_r0;
+
+        // Get the inverse exponential HLR, which is used for calculating stepK
+        _inv_exp_re = 1./(1.6783469900166605*_r0);
+
+        _shootnorm = _flux * _info->getXNorm(); // For shooting, we don't need the 1/r0^2 factor.
+        _xnorm = _shootnorm * _inv_r0_sq;
+        dbg<<"norms = "<<_xnorm<<", "<<_shootnorm<<std::endl;
+
+        // For small k, we can use up to quartic in the taylor expansion of both terms
+        // in the calculation.
+        // This is acceptable when the next term is less than kvalue_accuracy.
+        // (35/16 + 31/15120 pi/2*h*sin(i)/r) * (k^2*r^2)^3 = kvalue_accuracy
+        // This is a bit conservative, note, assuming kx = 0
+        _ksq_min = std::pow(this->gsparams->kvalue_accuracy /
+                            (35./16. + 31./15120.*_half_pi_h_sini_over_r), 1./3.);
```

Comment:
It might help to look at the kValueHelper function in SBInclinedExponential.cpp. In the inclined exponential case, _ksq_min is used as the limit for k in both res_base and res_conv, so when calculating _ksq_min, we need to take this into account so that the sum of errors introduced through this simplification won't exceed kvalue_accuracy.

In the inclined sersic case, res_base is calculated by the SersicInfo class, which has its own internal ksq_min (determined in a similar way). If we were to calculate ksq_min for the SBInclinedSersic class as if it were only relevant for res_conv, then we'd be allowing for res_conv to have error in it of up to kvalue_accuracy. But at the same time, res_base (calculated from the SersicInfo class) also might have error up to kvalue_accuracy, and so res_base*res_conv could have error up to double(ish) kvalue_accuracy, which we don't want.